### DISCUSSION TITLE
Levels and Prestige

### DISCUSSION THREAD

**Discussion by VV-LX** [ppy/osu#28838] [Ideas]
### Prestige 
A full Level system that provides rewards (banners, borders, titles, supporter, [coins](https://github.com/ppy/osu/discussions/28837)) at Milestones (5, 10, 20, etc). Upon reaching a certain Level (ex. 100), the levels can be reset. Besides giving EEEEEEEEEEEEEEE a number of similar length beside their name, Prestige is meant to replace the massive flat XP requirements after LVL 100 in a more intuitive and striking way. 

Every Prestige can also reflect in a significant profile presence (for E^16, a Prestige DCCXXVII with rainbow and flame animations shooting out)

### XP
XP should be counted/given separately when playing with Modifiers. Alone, this would be a meaningless way to inflate ranked score, but in tandem with Milestone Rewards, this becomes an incentive to experiment and play with more obscure Mods (EZ, FL, and Lazer Exclusives)

### Prestige Trials
 Before resetting Levels, players must complete a set of requirements (likely involving a set of achievements/medals) each reset giving a Prestige Rank (I, II, III…). This could even be tailored to each player through choices they can actively make, branching out in different ways to truly reflect the nature of skill progression.

This will use the XP curve that the current system does for 0-99, but keeping the same base curve for all levels, multiplied by each Prestige Rank.

**Comment by BlockHead007** [comment]
#27686 under level system improvements
Looks like it will be fully reworked in the future

**Reply by Stoppedpuma** [reply]
More precisely, #17124 is the thread discussing reworking user levels.

**Reply by VV-LX** [reply]
It definitely makes Levels move towards a holistic direction, but I was more focused on how Levels would end up being presented regardless (and how intuitively "special" it would feel based on how it looked). 

I also noticed #17124 discussed incorporating Medals into the XP system. The idea was to make Medals more relevant to players, but I think XP is a generic way to reward a player's dedication to pursuing a specific medal. 

A couple of maps have their own "banners"; couldn't we start by letting those banners be the design for medals, or even letting players "hang" those banners on their profile in a specific area (rather than the me! section).

"Trials" are also a way to make Medals/specific Achievements more approachable/enticing by officially incorporating the achievement hunting niche of osu.

**Reply by Stoppedpuma** [reply]
> A couple of maps have their own "banners"

Could you explain what this means? My best guess is backgrounds but every beatmap has one?

**Reply by VV-LX** [reply]
Sure. Some maps have a person who took the extra time out of their day to not get recognized for their creation. This was probably the storyboarder.


https://osu.ppy.sh/beatmapsets/1124097#osu/2348896
![image](https://github.com/user-attachments/assets/6acfb499-0c2d-43d3-b5e7-53b1a24401bf)

https://osu.ppy.sh/beatmapsets/1777727#osu/3640549
![image](https://github.com/user-attachments/assets/39ee6ee1-be0e-42ea-99c2-6d5de62e8a52)


These are Banners you can find in the descriptions of the beatmaps but I think adapting a couple of Storyboards would also work. Just a BG wouldn't be very unique (since most BG's can be used by anyone) but Storyboards/custom Banners are specifically identifying. 

**I'd imagine iconic maps would first be selected for this kind of reward, each reward tagged with a date of when it was obtained.** Just imagine a date xx/xx/xxxx in small font in an unobtrusive location those images.

Having the option to earn the right to display them on your profile in a unique location (such as underneath your webpage) would make them pretty cool.

**Reply by Stoppedpuma** [reply]
Thanks! I think this makes sense for maps with medal rewards, the problem is implementation. I think just letting the user throw them on their profile might cause confusion with profile badges. Maybe having them as unlockable profile covers would work better?

**Reply by VV-LX** [reply]
I think profile covers would be too restrictive with how many of these some players would accrue. I imagined them given in a size similar to Medals, and there being a proper area to display them where you can properly pick and choose which Icons to put up. 


A sidebar could be placed under the user info but above me! for users to put up their Icons.
![image](https://github.com/user-attachments/assets/820c69d0-f241-47b3-924d-28d017dfab71)

I liked Profile Colors with how they actually change the entire feel of the page. Could Covers do the same? Instead of being restricted to a small rectangle all the way at the top of the user page, how about making the userpage transparent and they lay underneath?

**Reply by VV-LX** [reply]
What are your thoughts on making the user's Level appearance more graphic/animated? It's hard to actually notice the Border/Level Changes when looking at a user's profile page, so something like a Border _around_ a User's profile (Profile Border) would be more eye catchy (like what LoL does)

**Reply by Stoppedpuma** [reply]
> and there being a proper area to display them

I think this still feels out of place in the design.

![image](https://github.com/user-attachments/assets/3a55efcd-4f32-49fe-be39-19818ade1d41)

> how about making the userpage transparent 

This is one of my larger dislikes in "game design" and other various places that implement it since it's almost never done in a good / correct way. Not only do I think these designs look worse majority of the time, I also find this moderately hard to read from any distance beyond 5-10 metres, add into the equation smaller screens such as laptops or phones and you can probably see how this would be a downhill slope. The design below is probably the best example I could make in-terms of readability within five minutes (and probably one of the better ones I've seen in terms of readability to be honest). Additional problems exist with this as well, most notable includes pictures as this require downloading them, not everyone has fast connections and may even struggle loading seasonal backgrounds. This also is a pretty jarring feel to go from a easy on the eyes design to cluster that some images could be taking up most of your screen, this also leads to contrast issues.

![transparent](https://github.com/user-attachments/assets/3eb26998-c385-4429-9b8b-566db8cccc58)

> Level appearance more graphic/animated

You might find interest in this, this is the upcoming redesign for levels.

![image](https://github.com/user-attachments/assets/52dc0cab-b4a7-4344-aef7-45ea6c41137e)

Additionally medals which shows making the above even harder.

![image](https://github.com/user-attachments/assets/06a521d8-9de3-4f00-aab1-c075a9e3b139)

> like what LoL does

This seems to be most of what you're basing these discussions on, I personally see 90% of Riots designs and implementations as failures so take that as you will. I don't think making osu! a clone of other "modern games"  behaviourally is a beneficial thing.

**Reply by VV-LX** [reply]
> I think this still feels out of place in the design.

Are there any plans to make a redesign for more sections on the Profile page? I think making each part of the profile (me!, Ranks, etc) sectioned into Tabs instead of a long scroll could make it easier to digest the Background.


> I also find this moderately hard to read from any distance beyond 5-10 metres

Do you think it would look better if the background was more opaque/dimmed (like it was behind gray glass)? The top, smaller image might also feel redundant, so it could be removed to make space for badges or something else. 
Maybe I just have bad vision, but I don't play/read stuff at a distance of 5m (I can't read my current osu! profile from that distance)


> You might find interest in this, this is the upcoming redesign for levels.

Titles are sick. Gradient-wise I think there could be a difference between lvl 80 and lvl 100. I was also thinking the _inside_ of the Level icon could be filled with some design.


> I don't think making osu! a clone of other "modern games" behaviourally is a beneficial thing.


What are your thoughts on something like the Prestige system (visually and progression-wise) on CoD?

**Reply by VV-LX** [reply]
For osu, I was imagining Prestige to look like this
![image](https://github.com/user-attachments/assets/e18fb347-2726-4174-926a-795845e2e6e0)

**Reply by Stoppedpuma** [reply]
> Are there any plans to make a redesign for more sections on the Profile page?

![image](https://github.com/user-attachments/assets/7af018bd-3000-4003-bb70-5d572413fa26)

> Do you think it would look better if the background was more opaque/dimmed (like it was behind gray glass)?

Currently profile colours can only be changed with Hue, Saturation and Brightness aren't options yet and I don't know if they ever will be since this allows users to create "blank profiles" by making everything white.

> The top, smaller image might also feel redundant, so it could be removed to make space for badges or something else.

These are profile covers, they have been here for years, Removing them is going to be a hard no.

> Gradient-wise I think there could be a difference between lvl 80 and lvl 100

Gradients follow this order, they are awarded at the positions in the picture under it.

![image](https://github.com/user-attachments/assets/f22ec964-e845-4174-a145-3f7bb40356e2)
![image](https://github.com/user-attachments/assets/fcf18a65-35c6-493b-addd-9d32598764ba)


> What are your thoughts on something like the Prestige system (visually and progression-wise) on CoD?

I don't see a purpose for prestiges.

**Reply by VV-LX** [reply]
> since this allows users to create "blank profiles" by making everything white.

I meant if you could add some opacity/dim to the mockup you made (Like how BG Dim is in-game)
![image](https://github.com/user-attachments/assets/0839a240-f39c-4b22-ab60-f3a538024f62)

> These are profile covers, they have been here for years, Removing them is going to be a hard no.

I meant removing the cover in that mockup because it would be redundant considering there was a larger version of it in the background. In essence, I was wondering if Covers would have a better, more prominent/larger display location on a User's Profile.

> Gradients follow this order
Does that mean Lvl 20 and every Lvl after it shares the same gradient? I think having more variety, especially for leaderboard, would make it feel more special.

> I don't see a purpose for prestiges.

It could be nice for bragging rights/give depth to Levels

**Reply by Stoppedpuma** [reply]
> I meant if you could add some opacity/dim to the mockup you made (Like how BG Dim is in-game)

This is under the assumption that this is user customisable, the same issues would occur. This also still ignores the issues with limited network connections and a huge one previously missed, user content being potentially disturbing or borderline 18+.

> I meant removing the cover in that mockup because it would be redundant considering there was a larger version of it in the background. In essence, I was wondering if Covers would have a better, more prominent/larger display location on a User's Profile.

You have to take the client into scale as well, currently they are only designed to show the main content and nothing bordering it. This adds to the overlay feel as well. Adding images here will not work out well as this takes away from that.

![image](https://github.com/user-attachments/assets/6cc4bf59-1960-4027-9ec8-e6e7b9be2bd1)

> It could be nice for bragging rights/give depth to Levels

I find this completely useless and nothing more than excess development for nothing. I think the border gradients works fine enough as a "progress" statement.

**Reply by VV-LX** [reply]
> user content being potentially disturbing or borderline 18+.

Is this also a problem with user profile pages/profile covers? If it is still a concern, couldn't the Backdrop picture be limited to a specific selection (like the Banners suggested earlier).


> I find this completely useless and nothing more than excess development for nothing.

That's fair. A question though; why are border gradients passable but a changeable icon inside that border (just like a medal) not?
I feel the setup is already there going by this picture
![image](https://github.com/user-attachments/assets/726497c4-2068-46c5-983e-2118388774c5)
The Level number is sitting on top of a colored Shield. In the web page you showed, there was just a Level. Couldn't it have a different symbol/shape for each Level milestone?


Besides that, I'd like to know if there are any plans to increase profile customizability, even on a base level (like with Profile Colors)

**Reply by Stoppedpuma** [reply]
> Is this also a problem with user profile pages/profile covers?

There's a lot more room in the approach you're wanting. I've already seen my fair share of people with "cropped breast profile pictures" if that's a correct term? I think you can see how this can be a controversial change as people already complain about nsfw art in the art contests.

> If it is still a concern, couldn't the Backdrop picture be limited to a specific selection (like the Banners suggested earlier).

This requires either art contest winners or official designs. As of now art contest winners have temporary banners which exist for two weeks.

> That's fair. A question though; why are border gradients passable but a changeable icon inside that border (just like a medal) not?

Border as in level border:

![image](https://github.com/user-attachments/assets/68660505-4e4f-42d8-97c8-1db59467a028)

Putting an image inside of this is going to look cluttered in my opinion.

> The Level number is sitting on top of a colored Shield. In the web page you showed, there was just a Level. Couldn't it have a different symbol/shape for each Level milestone?

Those aren't levels, they are GMT badges for how many years you've been one. The image was meant as a visual of the gradients and not current design.

> Besides that, I'd like to know if there are any plans to increase profile customizability, even on a base level (like with Profile Colors)

As of now, lazer client skinning. Any other additions would probably come after the profile page is fully updated to the new designs.
